apps/web/src/hooks/useDataRef.ts

```diff
@@ -0,0 +1,8 @@
+import { useRef } from 'react';
+
+export const useDataRef = <T>(data: T) => {
+  const ref = useRef<T>(data);
+  ref.current = data;
+
+  return ref;
+};
```

Comment:
the new hook that is just a ref that can hold anything and it can be used for ex. in the `useEffect` without defining it in the deps array